### ISSUE TITLE
Additional Restrictions on top of day and time restrictions.

### ISSUE THREAD

**Issue by papakaliati** [jamesmh/coravel#22]
Add the possibility Restrictions on top of day and time restrictions to provide more control.

**Comment by jamesmh** [comment]
Can you be more specific? List out the exact methods you think would be beneficial?

Thanks @papakaliati !

**Comment by papakaliati** [comment]
An optional CustomRestrictions based on an Func<bool> provided by the user that will be checked alongside Date and Time restrictions.

**Comment by jamesmh** [comment]
I closed the PRs. I'm actively overhauling the scheduler feature since it's was more-or-less a "get this out the door quick" to see how the community would receive it.

Implement the `When()` restriction will be much easier once I'm finished. I'll open a new issue and assign you to it. Once the next release is done I'll let you know and you can contribute that feature.

Thanks again for your effort and interest!

**Comment by jamesmh** [comment]
Leave a comment here and I'll be able to assign you 😄 

https://github.com/jamesmh/coravel/issues/30#issue-352221748
